wagtail/admin/tests/test_edit_handlers.py

```diff
@@ -1061,13 +1061,34 @@ def setUp(self):
         self.reply_1 = CommentReply.objects.create(comment=self.comment, text='reply_1', user=self.other_user)
         self.reply_2 = CommentReply.objects.create(comment=self.comment, text='reply_2', user=self.commenting_user)
 
+    def tearDown(self):
+        set_default_page_edit_handlers(Page)
+
     def test_comments_toggle_enabled(self):
         """
         Test that the comments toggle is enabled for a TabbedInterface containing CommentPanel, and disabled otherwise
         """
         self.assertTrue(self.tabbed_interface.show_comments_toggle)
         self.assertFalse(TabbedInterface([ObjectList(self.event_page.content_panels)]).show_comments_toggle)
 
+    @clear_edit_handler(Page)
```

Comment:
You could make this happen automatically using Django's ``setting_changed`` signal. For example: https://github.com/wagtail/wagtail/blob/main/wagtail/core/utils.py#L290